src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -558,6 +580,122 @@ def __init__(self, launch_string):
         super().__init__(details)
 
 
+class SlurmFuture:
```

Comment:
Closely follow the [Future interface](https://docs.python.org/3/library/concurrent.futures.html#future-objects)